### PR TITLE
typescript: introduce and port

### LINKED ISSUE #11: Expose TypeScripts types
It doesn't mean that the project has to be written in TypeScript, an `index.d.ts` file at the root would be a good start.

https://github.com/Microsoft/dts-gen
https://stackoverflow.com/a/47746553
https://stackoverflow.com/a/12695001
http://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html
http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

### PR DESCRIPTION
**Pull request by bjornars**
This resolves #11.

### CODE REVIEW

**Comment by bjornars** [comment]
I did notice just now that since the README talks about `npm`, I have used `npm` for dependencies, leaving `yarn.lock` in a probably confused state.

File: src/xapi/index.ts [hunk 1]
```diff
@@ -145,15 +151,16 @@ export default class XAPI extends EventEmitter {
...
     const method = `xCommand/${apiPath}`;
-    const executeParams = body === undefined ? params : Object.assign({ body }, params);
+    const executeParams =
+      body === undefined ? params : Object.assign({ body }, params);
     return this.execute(method, executeParams);
   }
 
   /** @private */
-  handleResponse(response) {
+  public handleResponse(response: XapiResponse) {
```
Comment (myme) [inline src/xapi/index.ts]: `private`?

File: src/xapi/index.ts [hunk 2]
```diff
@@ -173,7 +180,7 @@ export default class XAPI extends EventEmitter {
   }
 
   /** @private */
-  nextRequestId() {
+  public nextRequestId() {
```
Comment (myme) [inline src/xapi/index.ts]: `private`?

File: src/xapi/types.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  params?: any;
+}
+
+export interface XapiResponse {
+  id: string;
+  method: string;
+  params: any;
+  result: XapiResult;
+  error: XapiError;
+}
```
Comment (myme) [inline src/xapi/types.ts]: Some of these properties are mutually exclusive. I think in `json-rpc` you either get a `result` or an `error`, depending on the response being a success or a failure. I'm not looking at the spec currently, but I think `params` is optional too.

**Comment by myme** [comment]
I'm declining this PR in favor of #36.

**Pull request closed without merging**